sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -424,12 +424,16 @@ abstract class NewRule {
     /**
      * The optional description, in a restricted Markdown format, has no max length. It's exclusive with HTML description
      * (see {@link #setHtmlDescription(String)})
+     * @deprecated since 9.6, use {@link #addDescriptionSection(RuleDescriptionSection)} instead
      */
+    @Deprecated
```

Comment:
@dbmeneses [sonar]([https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.api.plugin%3Aso[…]est=19&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.api.plugin%3Asonar-plugin-api&open=AYB1P3Eo0neSvZdO1PFE&pullRequest=19&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)) wants me to set both `since` and `forRemoval` fields of the deprecated `@Deprecated` ; the `since` is easy... but what about the `forRemoval`?

Should we declare that version '10' will be the fix version? should I simply fix the issues as won't fix?

Thanks for your feedback.

Reply:
according to JDK 9[ release notes](https://www.oracle.com/java/technologies/javase/9-deprecated-features.html): `The forRemoval=true text indicates that a deprecated API might be removed from the next major release. The forRemoval=false text indicates that a deprecated API is not expected to be removed from the next major release but might be removed in some later release.`
We don't have a deprecation policy well defined for the API yet, but I think in general we can mark things for removal "in the next major release" (whatever that means), even if that's not necessarily the case. WDYT?

Reply:
it's the first time I see this issue BTW (probably because we were targeting java 8 until recently)

Reply:
I think it's indeed okay to mark it as "forRemoval".  The worse that can happen is that we actually don't remove it in next major release, which is ok :)